nengo_gui/components/htmlview.py

```diff
@@ -10,31 +12,50 @@ class HTMLView(Component):
 
     def __init__(self, obj):
         super(HTMLView, self).__init__()
-        self.obj = obj
-        self.obj_output = obj.output
         self.data = collections.deque()
 
+        self.obj = obj
+        self.obj_output = self.obj.output
+        self.callable_html = callable(obj.output._nengo_html_)
+        if self.callable_html:
```

Comment:
I tried this out. In some ways it's cleaner, but the inheritance does add it's own form of cognitive complexity. Frankly, I'm just not sure if this class is large enough to really benefit from inheritance.

And now, I'm getting an error: ``ConfigError: Type 'CallableHTMLView' is not set up for configuration. Call configures('CallableHTMLView') first.``

I've put the code in the ``html-function-refactor`` branch. In the spirit of reviewers making the changes, you guys are welcome to run with that and try to figure out the error. But I don't have time to sink into that, and I think that what I have here is fine (or at least not worth the effort).